### PR TITLE
Update OpenTitan EDN registers and `SW_CMD_STS` behavior

### PR DESCRIPTION
**Pull request by luismarques**

### CODE REVIEW

File: hw/opentitan/ot_edn.c [hunk 1]
```diff
@@ -231,14 +234,22 @@ struct OtEDNState {
     uint32_t *regs;
 
     unsigned reseed_counter; /* track remaining requests before reseeding */
-    bool last_cmd_failed; /* status of the last CSRNG command */
+    uint8_t last_cmd_status; /* status of the last CSRNG command */
```
Comment (AlexJones0) [inline hw/opentitan/ot_edn.c]: Is there a reason that this has type `uint8_t` and not `OtCSRNGCmdStatus`?

**Review by AlexJones0 — APPROVED** [review]
Thanks @luismarques, these changes all appear to match the latest EDN docs to me.

File: hw/opentitan/ot_edn.c (hunk 1 above)
Comment (luismarques) [reply hw/opentitan/ot_edn.c]: My preferred way to do that would be to declare the enum using `enum OtCSRNGCmdStatus : uint8_t`, and then the `last_cmd_status` as a `OtCSRNGCmdStatus`. But that requires C23. I don't mind changing `last_cmd_status` to `OtCSRNGCmdStatus`, though it takes a few extra bytes, and the actual status field is only 3 bits or something like that.

Comment (AlexJones0) [reply hw/opentitan/ot_edn.c]: That makes sense, thanks for the explanation. I think this is fine to stay as is - the status field should not really ever have more than 256 values.

File: hw/opentitan/ot_edn.c [hunk 2]
```diff
@@ -1071,12 +1084,15 @@ static uint64_t ot_edn_regs_read(void *opaque, hwaddr addr, unsigned size)
     case R_ERR_CODE_TEST:
         val32 = s->regs[reg];
         break;
-    case R_SW_CMD_STS:
-        val32 =
-            FIELD_DP32(0, SW_CMD_STS, CMD_STS, (uint32_t)s->last_cmd_failed);
-        val32 = FIELD_DP32(val32, SW_CMD_STS, CMD_RDY,
-                           (uint32_t)ot_edn_is_cmd_rdy(s));
+    case R_SW_CMD_STS: {
+        uint32_t rdy = ot_edn_is_cmd_rdy(s, false);
```
Comment (rivos-eblot) [inline hw/opentitan/ot_edn.c]: Some explicit cast (bool -> uint32_t) would be better.

File: hw/opentitan/ot_edn.c (hunk 1 above)
Comment (rivos-eblot) [reply hw/opentitan/ot_edn.c]: We had the very same requirement this very week :-)
Following a review on our side that I have yet to push, could you add a comment about the use of this 8-bit value? BTW C99 enum are (signed) integers, so I think `int8_t` would be better.

Comment (rivos-eblot) [reply hw/opentitan/ot_edn.c]: and maybe moved it towards line 199 along with other enums as it would introduce the field in the structure.

**Review by rivos-eblot — CHANGES_REQUESTED** [review]

File: hw/opentitan/ot_edn.c (hunk 1 above)
Comment (rivos-eblot) [reply hw/opentitan/ot_edn.c]: > We had the very same requirement this very week :-) 

https://github.com/lowRISC/qemu/pull/87/commits/a4ebcfa7fd7493b343541b7d51621dbfa6e3a21d

Comment (luismarques) [reply hw/opentitan/ot_edn.c]: > We had the very same requirement this very week :-) Following a review on our side that I have yet to push, could you add a comment about the use of this 8-bit value? BTW C99 enum are (signed) integers, so I think `int8_t` would be better.

OK, I just typed the member variable with the enum type. That avoids the need for the comment, sidesteps the `uint8_t` vs `int8_t` issue, and when QEMU later adopts C23 we can just type the enum as `uint8_t`.

**Review by rivos-eblot — APPROVED** [review]
LGTM.

**Pull request merged**